DependencyInjection/SmartboxIntegrationFrameworkExtension.php

```diff
@@ -229,45 +235,49 @@ protected function loadQueueDrivers(ContainerBuilder $container)
                     break;
 
                 case 'amqp':
-                    if (!class_exists('AMQPConnection')) {
-                        throw new \RuntimeException('You need the amqp extension to use AMQP driver.');
+                    $this->useAmqp = true;
+
+                    $this->drivesThatUseAMQP[] = $driverName;
+
+                    if (!class_exists(AMQPStreamConnection::class)) {
+                        throw new \RuntimeException('You need the amqp extension to use PHP-AMQP-LIB driver.');
                     }
 
                     if (empty($driverConfig['connections'] ?? [])) {
-                        throw new \InvalidArgumentException('You need to define at least one connection to use the AMQP driver.');
+                        throw new \InvalidArgumentException('You need to define at least one connection to use the PHP AMQP lib driver.');
                     }
 
-                    $queueManager = $container->findDefinition('smartesb.amqp.queue_manager');
+                    $driverDef = new Definition(PhpAmqpLibDriver::class);
+
                     foreach ($driverConfig['connections'] as $index => $uri) {
                         $connection = parse_url($uri);
-                        $connectionId = "$driverId.connection.$index";
-
-                        $container->register($connectionId, 'AMQPConnection')->setArguments([[
-                            'host' => $connection['host'],
-                            'port' => $connection['port'] ?? 5672,
-                            'vhost' => trim($connection['path'] ?? '', '/'),
-                            'login' => $connection['user'],
-                            'password' => $connection['pass'],
-                        ]]);
 
-                        $queueManager->addMethodCall('addConnection', [new Reference($connectionId)]);
+                        $driverDef->addMethodCall('setPort', [$driverConfig['port'] ?? AsyncQueueDriverInterface::DEFAULT_PORT]);
+                        $driverDef->addMethodCall('configure', [
+                            $driverConfig['host'],
+                            $driverConfig['username'],
+                            $driverConfig['password'],
+                            $driverConfig['vhost']
```

Comment:
Right now this driver can't be setup with multiple amqp hosts because the fields are the same for both connections. What was happening in the past with Gary's implementation is that he was parsing the connection uri and from there he was taking the different configuration